### PR TITLE
Adds support for debugging in VS Code

### PR DESCRIPTION
**Pull request by simongdavies**
Adds tasks, scripts and launch configurations to enable support for debugging in Visual Studio Code.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -32,3 +32,45 @@ Here are the most common Makefile tasks:
...
+* `PORTER_PLUGIN_WORKING_DIRECTORY` should be the local path to the source code for the plugin being debugged. (e.g. for the Azure plugin this value would be `<PATH>/porter-azure-plugins/cmd/azure`
+
+Porter will only attach the debugger to an existing binary in **~/.porter/plugins**. It is therefore necessary to build and install the plugin before debugging, use `make debug` to do this.
+
+### Debugging Scenarios
+
+There are 2 different debugging scenarios:
+
+#### 1. Debug the plugin only. 
+1. Run `make build-debug`
```
Comment (vdice) [inline CONTRIBUTING.md:51]: Should this target be `build-for-debug`? Or perhaps `debug`?  Same on line 58.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -32,3 +32,45 @@ Here are the most common Makefile tasks:
...
+1. Set the environment variables as described above.
+1. Set the [appropriate configuration in config.toml](https://porter.sh/plugins/azure/).
+1. Run porter with a command that invokes the plugin e.g. `porter --tag getporter/plugins-tutorial:v0.1.0 --debug -c plugins-tutorial`
```
Comment (vdice) [inline CONTRIBUTING.md:54]: This command needs an action, right?  Maybe `porter install ...`?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -32,3 +32,45 @@ Here are the most common Makefile tasks:
...
+1. Connect the debugger to the delve API (by default this will be listening at 127.0.0.1:2345), e.g using delve `dlv connect 127.0.0.1:2345`
+
+#### 2. Debug porter and the plugin simultaneously.
+1. Run `make build-debug`
+1. Set the environment variables as described above.
+1. Set the [appropriate configuration in config.toml](https://porter.sh/plugins/azure/).
+1. Run porter under the debugger with a command that invokes the plugin e.g. `dlv exec <path-to-porter> -- --tag getporter/plugins-tutorial:v0.1.0 --debug -c plugins-tutorial` . Note that for this to work porter should be compiled so that the debugger can launch the binary successfully, the simplest way to do this is to build with no options.
```
Comment (vdice) [inline CONTRIBUTING.md:61]: The porter command portion needs an action, right?  Maybe `install`?

**Review by vdice — COMMENTED** [review]
Followed the 'Debug the plugin only' steps and appears to look good!

One Q: Are these lines expected?

```
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3bb884) of inlined call at 0x3b6fe5
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d1e99) of inlined call at 0x3b7b57
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d1e99) of inlined call at 0x3b7dbb
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d1e99) of inlined call at 0x3b7f37
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b80ad
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b8125
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b8b60
2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b8c30
....
```

**Comment by simongdavies** [comment]
> Followed the 'Debug the plugin only' steps and appears to look good!
> 
> One Q: Are these lines expected?
> 
> ```
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3bb884) of inlined call at 0x3b6fe5
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d1e99) of inlined call at 0x3b7b57
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d1e99) of inlined call at 0x3b7dbb
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d1e99) of inlined call at 0x3b7f37
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b80ad
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b8125
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b8b60
> 2020-09-09T16:26:25-06:00 warning layer=debugger reading debug_info: could not find abstract origin (0x3d205d) of inlined call at 0x3b8c30
> ....
> ```

I think this happens when the binary being debugged is optimised, I think building without `-ldflags` should prevent this

**Review by vdice — APPROVED** [review]

**Pull request merged**